docs/USAGE-GUIDE.md

```diff
@@ -1777,6 +1777,164 @@ handle. The handle is considered invalid after **s2n_connection_free** is used.
+PSK object created by this API.
+
+**s2n_psk_free** frees the memory associated with the **s2n_psk** object.
+
+Use **s2n_psk_set_identity** and **s2n_psk_set_secret** to set the pre-shared key identity and secret after creating the external pre-shared key object. Use **s2n_psk_set_hmac** to override the default hash algorithm.
+
+**s2n_psk_set_identity** sets the PSK identity for a given s2n_psk object. The PSK identity is a unique identifier for the pre-shared secret.
+The Pre-shared key identity is transmitted over the network unencrypted and is a non-secret value. Do not include confidential information in the pre-shared key identity. Note that the secret value is copied into s2n-tls memory and the user is responsible for freeing the memory associated with the **identity** input.
+Be sure to call **free(secret)** after this API call.
+
+**s2n_psk_set_secret** sets the out-of-band/externally provisioned secret for a given PSK. Note that the secret value is copied into s2n-tls memory and the user is
+responsible for freeing the memory associated with the **secret** input. Be sure to call **free(secret)** after this API call.
+
+**s2n_psk_set_hmac** sets the PSK hash algorithm for a given PSK.
+The supported PSK hash algorithms are listed in the **s2n_psk_hmac** enum. Note that in versions >= TLS1.3 the cipher suite selected has the same Hash alogrithm
+as the PSK. Use this API to override the default PSK Hash Algorithm value of **S2N_PSK_HMAC_SHA256**. From [RFC8446 Section-2.2](https://tools.ietf.org/html/rfc8446#section-2.2) when PSKs are provisioned out-of-band the KDF hash algorithm to be used with the PSK MUST also be provisioned. This API helps setting the KDF hash algorithm to be used with the PSK.
+
+After adding an identity and secret to the PSK object, append the PSK to the connection using **s2n_connection_append_psk**. This API sets up the list of PSKs on the connection that will later be used by the PSK extension to select a PSK for the connection. If a PSK with a duplicate identity is found, an error is returned and the PSK is not added to the list. Note that the PSK object is copied to the connection and the user is responsible for freeing the memory associated with the s2n_psk object. Use **s2n_psk_free** to free the memory associated with the external PSK object after using this API.
+
+**s2n_config_set_psk_mode** sets the PSK mode on the s2n config object. The enum **s2n_psk_mode** lists the supported PSK modes. Use this API prior to the ClientHello to set the PSK mode for the connection. This API can also be used to switch between the different PSK modes.
+
+**s2n_connection_set_psk_mode** sets the PSK mode on the s2n connection object. The supported PSK modes are listed in the enum **s2n_psk_mode** within the **api/s2n.h** file.
+This API overrides the PSK mode for the connection and will continue to use the existing PSK mode even when a new config is set for the connection.
```

Comment:
You should describe actual usage cases for these two. Customers are unlikely to actually need to call them.